src/app/admin/accessibility-management/accessibility-management.component.ts

```diff
@@ -7,12 +9,19 @@ import { Component, OnInit } from '@angular/core';
 })
 export class AccessibilityManagementComponent implements OnInit {
 
-  constructor() {
-  }
-
-  ngOnInit(): void {
-  
-  }
+    isAllowed: boolean = false;
 
+    constructor(private authService: AuthService,
+                private routerService: Router) {
+    }
 
+    ngOnInit(): void {
+    this.authService.getUserProfile().then((resp) => {
```

Comment:
Good. But let's try to keep things DRY (don't repeat yourself): move this app logic in one 'helper' class or a static method that should: 
- receive the allowed roles as param 1
- redirect to /admin in case the user was not allowed in